### PR TITLE
mednafen: 0.9.47 -> 0.9.48, disable PIC, unversioned docs

### PR DESCRIPTION
**Pull request by lukateras**
###### Motivation for this change

Version bump. HTTP -> HTTPS in `meta.homepage`, simpler `postInstall`.
Mednafen doesn't want to be compiled with PIC, so now it's disabled:

```
../../include/mednafen/types.h:28:4: warning: #warning "Compiling with position-independent code generation enabled is not recommended, for performance reasons." [-Wcpp]
   #warning "Compiling with position-independent code generation enabled is not recommended, for performance reasons."
    ^~~~~~~
```

This also includes using `$out/share` directory that is similar to what other distros use, i.e. no version in the directory name.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/emulators/mednafen/default.nix [hunk 1]
```diff
@@ -1,36 +1,30 @@
-{ stdenv, fetchurl, pkgconfig
-, libX11, mesa, freeglut
-, libjack2, libcdio, libsndfile, libsamplerate
-, SDL, SDL_net, zlib
-}:
+{ stdenv, fetchurl, pkgconfig, libX11, mesa, freeglut, libjack2,
+, libcdio, libsndfile, libsamplerate, SDL, SDL_net, zlib }:
```
Comment (joachifm) [inline pkgs/misc/emulators/mednafen/default.nix]: Syntax error here

Comment (yegortimoshenko) [reply pkgs/misc/emulators/mednafen/default.nix]: Made a last-minute change, sorry about that. Fixed now.

**Review by AndersonTorres — APPROVED** [review]
It compiles OK here.

File: pkgs/misc/emulators/mednafen/default.nix [hunk 2]
```diff
@@ -1,36 +1,30 @@
...
     url = "https://mednafen.github.io/releases/files/${name}.tar.xz";
-    sha256 = "0flz6bjkzs9qrw923s4cpqrz4b2dhc2w7pd8mgw0l1xbmrh7w4si";
+    sha256 = "00i12mywhp43274aq466fwavglk5b7d8z8bfdna12ra9iy1hrk6k";
   };
 
-  buildInputs =
-  [ pkgconfig libX11 mesa freeglut libjack2 libcdio
-    libsndfile libsamplerate SDL SDL_net zlib ];
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ libX11 mesa freeglut libjack2 libcdio libsndfile libsamplerate SDL SDL_net zlib ];
```
Comment (AndersonTorres) [inline pkgs/misc/emulators/mednafen/default.nix]: Line surpassing the 80-column rule.

Comment (yegortimoshenko) [reply pkgs/misc/emulators/mednafen/default.nix]: Fixed.

**Pull request merged**